lib/components/pagination.vue

```diff
@@ -10,290 +10,404 @@
                :aria-setsize="numberOfPages"
                tabindex="-1"
-               @click.prevent="setPage($event, numberOfPages)"
-               @keydown.enter.prevent="setPage($event, numberOfPages)"
-               @keydown.space.prevent="setPage($event, numberOfPages)"
-            >{{numberOfPages}}</a>
+               @click.prevent="setPage($event, page.number)"
+               @keydown.enter.prevent="setPage($event, page.number)"
+               @keydown.space.prevent="setPage($event, page.number)"
+            >{{ page.number }}</a>
+        </li>
+
+        <!-- Last Ellipsis Bookend -->
+        <li v-if="showLastDots" class="page-item disabled hidden-xs-down" role="seperator">
+            <span class="page-link" v-html="ellipsisText"></span>
         </li>
 
+        <!-- Goto Next page -->
         <li v-if="isActive(numberOfPages) || disabled" class="page-item disabled" aria-hidden="true">
-            <span class="page-link">&raquo;</span>
+            <span class="page-link" v-html="nextText"></span>
         </li>
         <li v-else class="page-item">
-            <a role="button"
+            <a :role="buttonRole"
                class="page-link"
                :aria-label="labelNextPage"
                tabindex="-1"
                @click.prevent="setPage($event, currentPage + 1)"
                @keydown.enter.prevent="setPage($event, currentPage + 1)"
                @keydown.space.prevent="setPage($event, currentPage + 1)"
-            ><span aria-hidden="true">&raquo;</span></a>
+            ><span aria-hidden="true" v-html="nextText"></span></a>
+        </li>
+
+        <!-- Goto Last page -->
+        <li v-if="isActive(numberOfPages) || disabled" class="page-item disabled" aria-hidden="true">
+            <span class="page-link" v-html="lastText"></span>
+        </li>
+        <li v-else class="page-item">
+            <a :role="buttonRole"
+               class="page-link"
+               :aria-label="labelLastPage"
+               tabindex="-1"
+               @click.prevent="setPage($event, numberOfPages)"
+               @keydown.enter.prevent="setPage($event, numberOfPages)"
+               @keydown.space.prevent="setPage($event, numberOfPages)"
+            ><span aria-hidden="true" v-html="lastText"></span></a>
         </li>
 
     </ul>
 </template>
 
 <script>
-    function isVisible(el) {
-        return el && (el.offsetWidth > 0 || el.offsetHeight > 0);
-    }
+// Determine if an HTML element is visible - Faster than CSS check
+function isVisible(el) {
+    return el && (el.offsetWidth > 0 || el.offsetHeight > 0);
+}
 
-    export default {
-        data() {
-            return {
-                diff: 1,
-                showPrev: false,
-                showNext: false,
-                currentPage: this.value
-            };
+// Make an aray of N to N+X
+function makePageArray(startNum, numPages) {
+    return Array.apply(null, {length: numPages}).map(function(value, index){
```

Comment:
Yep.  rather than looping and pushing onto an initially empty array.  ES6 has `Array.fill(...)`, which makes an array of N length where each element is the same value... which would be nicer, but might require transpiling.

I _aquired_ that snippit from http://www.jstips.co/en/javascript/create-range-0...n-easily-using-one-line/